repro.in

```diff
@@ -223,9 +229,12 @@ function init_chroot(){
 
     # Prepare root chroot
     if [ ! -d "$BUILDDIRECTORY"/root ]; then
+        # Acquire lock early
+        mkdir -p "$BUILDDIRECTORY/root"
+        elock 9 "$BUILDDIRECTORY"/root.lock
+
```

Comment:
Good catch, but I think we should lock and then mkdir.

The mkdir couple of lines below can go now.